config/releases.exs

```diff
@@ -32,25 +32,34 @@ defmodule Environment do
     end
   end
 
-  def get_endpoint_static_url_config(nil), do: nil
-  def get_endpoint_static_url_config(""), do: nil
+  def get_endpoint_url_config(nil), do: nil
+  def get_endpoint_url_config(""), do: nil
+
+  def get_endpoint_url_config(url) do
+    uri = URI.parse(url)
 
-  def get_endpoint_static_url_config(host) do
     [
-      host: host,
-      scheme: get("STATIC_URL_SCHEME"),
-      port: get("STATIC_URL_PORT")
+      host: uri.host,
+      scheme: uri.scheme,
+      port: uri.port
     ]
   end
+
+  def get_uri_part(nil, _), do: nil
+  def get_uri_part("", _), do: nil
+
+  def get_uri_part(url, part) do
+    url
+    |> URI.parse()
```

Comment:
Is there any way we could avoid parsing the URL each time we call this function? Maybe we could parse (safely) both URLs (canonical and static) and _then_ pass those parsed URIs around?

Reply:
Let me try something, indeed.

Reply:
Actually, one of the issue here is that URI.parse(nil) isn't supported, therefore requires pattern matching prior to doing it. 🤔

Reply:
Yeah, we could have a function in `Environment` that parses them safely 😄

Reply:
I believe I've reached something interesting if you want to take a new look!